### PR TITLE
Add extra steps to delete resource handler trace

### PR DESCRIPTION
**Pull request by shyamjvs**
Based on https://github.com/kubernetes/kubernetes/issues/51899#issuecomment-329786131

cc @kubernetes/sig-scalability-misc @kubernetes/sig-api-machinery-misc @gmarek

### CODE REVIEW

File: staging/src/k8s.io/apiserver/pkg/endpoints/handlers/rest.go [hunk 1]
```diff
@@ -949,6 +949,7 @@ func DeleteResource(r rest.GracefulDeleter, allowsOptions bool, scope RequestSco
 				scope.err(err, w, req)
 				return
 			}
+			trace.Step("Read DeleteOptions body")
```
Comment (wojtek-t) [inline staging/src/k8s.io/apiserver/pkg/endpoints/handlers/rest.go]: Please delete it - it's not useful.

Comment (shyamjvs) [reply staging/src/k8s.io/apiserver/pkg/endpoints/handlers/rest.go]: Done.

**Pull request merged**